drivers/syslog/syslog_device.c

```diff
@@ -435,22 +435,44 @@ void syslog_dev_uninitialize(FAR struct syslog_channel_s *channel)
 {
   FAR struct syslog_dev_s *syslog_dev = (FAR struct syslog_dev_s *)channel;
 
-  /* Check if the system is ready */
+  /* Uninitializing a SYSLOG device should not take place within
+   * interrupt context.
+   */
 
-  if (syslog_dev_outputready(syslog_dev) < 0)
+  if (up_interrupt_context() || getpid() == 0)
     {
+      DEBUGASSERT(!up_interrupt_context() && getpid() != 0);
       return;
     }
 
-  /* Attempt to flush any buffered data */
+  /* The device cannot be unititialized while it is being
+   * initialized simultaneously.
+   */
+
+  DEBUGASSERT(syslog_dev->sl_state != SYSLOG_UNINITIALIZED &&
+              syslog_dev->sl_state != SYSLOG_INITIALIZING);
+
+  /* Attempt to flush any buffered data. */
 
   sched_lock();
   syslog_dev_flush(channel);
 
-  /* Close the detached file instance */
+  /* Close the detached file instance, and destroy the
+   * semaphore. These are both only created when the
+   * device is in SYSLOG_OPENED or SYSLOG_FAILURE
+   * state.
+   */
```

Comment:
```suggestion
  /* Close the detached file instance, and destroy the semaphore. These are
   * both only created when the device is in SYSLOG_OPENED or SYSLOG_FAILURE
   * state.
   */
```
nit: Improving the line occupancy.

Reply:
Done.